### DISCUSSION TITLE
what are 'unusual elements' ?

### DISCUSSION THREAD

**Discussion by bnjmnrsh** [mdn/mdn-community#425] [Browser compatibility data]
I started to open an issue on https://github.com/mdn/browser-compat-data, but I thought perhaps some discussion here would be a better way to start. 

Im currently researching behaviours around `fieldset` and `legend`, using `grid` and `flex` on the `fieldset` and `display: contents` on the nested `legend`. 
My initial tests with Safari 16.5 +voiceover appear to announce `fieldset` and `legend` just the same as when the legend is unstyled. However, this wasn't the result I was expecting as the caniuse compatibility data for [`css.properties.display`](https://caniuse.com/?search=display%3A%20contents), specifically this page: [CSS property: display: contents: Specific behaviour of unusual elements when display: contents is applied to them](https://caniuse.com/mdn-css_properties_display_contents_contents_unusual) seems to indicate it shouldn't work. 

The [MDN reference page](https://developer.mozilla.org/en-US/docs/Web/CSS/display) for `display` supports this and also makes an oblique reference to 'unusual elements' in the [Browser Compatibility table](https://developer.mozilla.org/en-US/docs/Web/CSS/display#browser_compatibility). Where Safari is highlighted as not supporting `display` properties on both 'unusual elements' AND `legend`. 

Now it's possible, even likely, I'm misinterpreting my own test results, as I'm, at best, a student of A11ty.  However, between MDN and caniuse resources, I feel there should be some definition as to what elements are classed as 'unusual", specifically in the context of `display: content`

Questions: 
- Shouldn't these resources point to the source of where the compatibility information is gathered from?
- ie what are these 'unusual elements', are fieldset and legend in that list?
- In writing this, I've just uncovered @aardrian 's [article](https://adrianroselli.com/2022/07/use-legend-and-fieldset.html#Display)  and feel that MDN and caniuse could be improved with some of his findings. 

A significant amount of discussion and work has already been done to document issues around `display: contents` that is worth referencing: 

CanIUse:
https://github.com/Fyrd/caniuse/issues/6469 
MDN Browser Compat Data:
https://github.com/mdn/browser-compat-data/issues?q=is%3Aissue+is%3Aopen+display+contents

Also:
https://adrianroselli.com/2022/07/use-legend-and-fieldset.html
https://adrianroselli.com/2022/07/its-mid-2022-and-browsers-mostly-safari-still-break-accessibility-via-display-properties.html
https://www.tpgi.com/fieldsets-legends-and-screen-readers-again/

Caniuse, receives its compatibility data from MDN: [`css.properties.display`](https://github.com/mdn/browser-compat-data/blob/main/css/properties/display.json), which is where any issue related to this should ultimately be logged.

**Comment by pepelsbey** [comment]
Thank you for starting the discussion. I’m closing this now due to inactivity, but if you think that’s a mistake, feel free to reply. I’d also suggest filing an issue directly in the [browser-compat-data repository](https://github.com/mdn/browser-compat-data/issues), which handles all the content in the notes for each compat entry. You’re also welcome to join the [MDN Discord](https://developer.mozilla.org/discord) if you’d prefer to chat with us more about it in sync.
